enable codeCoverage, setup codecov

michaelknoch opened the pull request:
<!-- Either add the type here or use a label and remove this part -->
**Type of change:** improvement

## Motivation
Trust is good, control is better

### Please check if the PR fulfills these requirements
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

* [ ] Self-review: I am confident this is the simplest and clearest way to achieve the expected behaviour
* [ ] There are no dependencies on other PRs or I have linked dependencies through Zenhub
* [ ] The commit messages are clean and understandable
* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Code runs on all relevant platforms (if major change: screenshots attached)

michaelknoch commented:
good question! @cshg 

I'm not sure what change @ephemer actually requested. If we still wan't to measure our code coverage it would make sense for me to merge it.

cshg commented:
@ephemer do you remember what changes you requested here?

ephemer commented on UIKitTests/UIProgressViewTests.swift:
```diff
@@ -0,0 +1,48 @@
+
+    func testCanSetProgressTintColor() {
+        progressView.trackTintColor = .green
+        XCTAssertEqual(progressView.backgroundColor, .green)
```
is this actually the case on iOS? I mean, do we run this test on both platforms?

ephemer commented on UIKitTests/UIProgressViewTests.swift:
```diff
@@ -0,0 +1,48 @@
+        progressView.layoutSubviews()
+
+        XCTAssertEqual(progressView.progress, newProgress)
+        XCTAssertNotNil(progressView.progressLayer._presentation)
```
we should always use the public API so we can test iOS too (_presentation is not public)

ephemer commented on UIKitTests/UIProgressViewTests.swift:
```diff
@@ -0,0 +1,48 @@
+    func testCanSetProgressWithAnimation() {
+        let newProgress: Float = 0.5
+        progressView.setProgress(newProgress, animated: true)
+        progressView.progressLayer.hasBeenRenderedInThisPartOfOverallLayerHierarchy = true
```
we should avoid this hack

ephemer commented on UIKitTests/UIProgressViewTests.swift:
```diff
@@ -0,0 +1,48 @@
+        progressView.setProgress(newProgress, animated: false)
+
+        XCTAssertEqual(progressView.progress, newProgress)
+        XCTAssertNil(progressView.progressLayer._presentation)
```
public API

ephemer commented on UIKitTests/DisplayLinkTests.swift:
```diff
@@ -0,0 +1,46 @@
+    var displayLink: DisplayLink!
+
+    override func setUp() {
+        DisplayLink.activeDisplayLinks = []
```
Not sure it makes sense to test this because it's our own implementation. What do you think? Is this just for code coverage?

ephemer commented on UIKitTests/DisplayLinkTests.swift:
```diff
@@ -0,0 +1,46 @@
+//
+
+import XCTest
+@testable import UIKit
```
Our test targets already import UIKit correctly so we should remove this

ephemer commented on README.md:
```diff
@@ -1,4 +1,5 @@
 [![CircleCI](https://circleci.com/gh/flowkey/UIKit-SDL.svg?style=svg)](https://circleci.com/gh/flowkey/UIKit-SDL)
```
Either way we'll have to update it here (either as resolving a conflict or just doing it here)

michaelknoch commented:
removed the tests @ephemer . you were right. I remember that i implemented those tests with the intention to improve coverage, not actually with the intention to prevent/fix bugs
